### DISCUSSION TITLE
Problem with pipeline plugin in Jenkins

### DISCUSSION THREAD

**Discussion by ensofter** [jenkinsci/bitbucket-push-and-pull-request-plugin#229] [Q&A]
Hello!

I encountered with problem - if i setup free-style job with yours plugin everything works as expected. But if i configure pipline job it doesn't work.

This is my pipeline configuration
![Screenshot (11)](https://user-images.githubusercontent.com/29799050/110099591-87260a80-7db2-11eb-97b4-0447078a5bf2.png)

This is trigers for new PR
![Screenshot (5)](https://user-images.githubusercontent.com/29799050/110043712-f3742000-7d58-11eb-89ee-f0d5858e514c.png)

I have webhook in bitbucket cloud relate to my repository and it returns to me status code 200
![Screenshot (6)](https://user-images.githubusercontent.com/29799050/110043920-51086c80-7d59-11eb-9c8a-971a37e6d434.png)
![Screenshot (7)](https://user-images.githubusercontent.com/29799050/110043988-71d0c200-7d59-11eb-8f81-2cf783188f4a.png)

When i do new PR i see in logs only this
![Screenshot from 2021-03-05 02-29-10](https://user-images.githubusercontent.com/29799050/110044730-adb85700-7d5a-11eb-9323-caed15cfa7f0.png)
In BitBucket Push and Pull Request Hook Log i see this
![Screenshot (8)](https://user-images.githubusercontent.com/29799050/110044863-eb1ce480-7d5a-11eb-98e3-8d3108ad36d8.png)

My jenkins file locate in root directory of repo
![Screenshot (9)](https://user-images.githubusercontent.com/29799050/110045017-199abf80-7d5b-11eb-8290-2090ac796993.png)
and contains this code
![Screenshot (10)](https://user-images.githubusercontent.com/29799050/110045054-29b29f00-7d5b-11eb-9c0d-90ff2d3c994d.png)

And i can't fix this. But free-style projects works fine.

Please, sorry for my english. I would be appriciate for any help.

**Comment by AckermannC** [comment]
I think (not 100% sure) the plugin is looking for an scm/checkout step inside the pipeline.

If it doesn't find one matching in a job it doesn't consider it.

But you have in Jenkins: "Manage Jenkins" -> "System Log" the chance to activate a more detailled logging:

If create a logger for io.jenkins.plugins.bitbucketpushandpullrequest with log level finest, you should be able to see how he is treating the job.

**Comment by macghriogair** [comment]
It should work without scm checkout step in the Jenkinsfile. 
But it would be good to see the detailed logs as @AckermannC suggested. Thanks!

**Comment by macghriogair** [comment]
Also, for debugging, please try to disable the other freestyle job if you haven't done so already.

**Comment by dmitrym0** [comment]
Having the same issue:

```
Mar 24, 2021 3:02:09 PM FINE io.jenkins.plugins.bitbucketpushandpullrequest.BitBucketPPRJobProbe
Considering candidate job Blackcomb PRs
Mar 24, 2021 3:02:09 PM FINE io.jenkins.plugins.bitbucketpushandpullrequest.BitBucketPPRJobProbe
Considering to poke Blackcomb PRs
```

> Consider to poke.. 

but it never does. Any idea what might be missing? I don't have any filtering setup.
